monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -478,3 +478,24 @@ def generate(self, output_folder=".", num_fold: int = 5):
                 gen_algo.export_to_disk(output_folder, name, fold=f_id)
                 algo_to_pickle(gen_algo, template_path=algo.template_path)
                 self.history.append({name: gen_algo})  # track the previous, may create a persistent history
+
+    def generate_with_customized_param(self, output_folder=".", num_fold: int = 5, **kwargs):
+        """
+        Automatically customize parameters of the bundle scripts/configs for each bundleAlgo based on gpus
+
+        Args:
+            output_folder: the output folder to save each algorithm.
+            num_fold: the number of cross validation fold
+        """
+        fold_idx = list(range(num_fold))
+        for algo in self.algos:
+            for f_id in ensure_tuple(fold_idx):
+                data_stats = self.get_data_stats()
+                data_src_cfg = self.get_data_src()
+                gen_algo = deepcopy(algo)
+                gen_algo.set_data_stats(data_stats)
+                gen_algo.set_data_source(data_src_cfg)
+                name = f"{gen_algo.name}_{f_id}"
+                gen_algo.export_to_disk(output_folder, name, fold=f_id, gpu_customization=True, **kwargs)
```

Comment:
Is it possible to optimize the param in fold 0 and then apply the optimized param set in other fold?

Reply:
yes. it would save a .yaml output for other folds. and the output can be reused in different GPU devices with similar memory.